packages/dcos-integration-test/extra/test_dcos_log.py

```diff
@@ -71,5 +72,145 @@ def test_log_proxy(cluster):
+    :param cluster: cluster fixture
+    :param task_name: task name
+    :param stream: use range or stream endpoint
+    :return: url to get the logs for a task
+    """
+    state_response = cluster.get('/mesos/state')
+    check_response_ok(state_response, {})
+
+    framework_id = None
+    executor_id = None
+    slave_id = None
+    container_id = None
+
+    state_response_json = state_response.json()
+    assert 'frameworks' in state_response_json, 'Missing field `framework` in {}'.format(state_response_json)
+    assert isinstance(state_response_json['frameworks'], list), '`framework` must be list. Got {}'.format(
+        state_response_json)
+
+    for framework in state_response_json['frameworks']:
+        assert 'name' in framework, 'Missing field `name` in `frameworks`. Got {}'.format(state_response_json)
+        # search for marathon framework
+        if framework['name'] != 'marathon':
+            continue
+
+        assert 'tasks' in framework, 'Missing field `tasks`. Got {}'.format(state_response_json)
+        assert isinstance(framework['tasks'], list), '`tasks` must be list. Got {}'.format(state_response_json)
+        for task in framework['tasks']:
+            assert 'id' in task, 'Missing field `id` in task. Got {}'.format(state_response_json)
+            if not task['id'].startswith(task_name):
+                continue
+
+            assert 'framework_id' in task, 'Missing `framework_id` in task. Got {}'.format(state_response_json)
+            assert 'executor_id' in task, 'Missing `executor_id` in task. Got {}'.format(state_response_json)
+            assert 'id' in task, 'Missing `id` in task. Got {}'.format(state_response_json)
+            assert 'slave_id' in task, 'Missing `slave_id` in task. Got {}'.format(state_response_json)
+
+            framework_id = task['framework_id']
+            # if task['executor_id'] is empty, we should use task['id']
+            executor_id = task['executor_id']
+            if not executor_id:
+                executor_id = task['id']
+            slave_id = task['slave_id']
+
+            statuses = task.get('statuses')
+            assert isinstance(statuses, list), 'Invalid field `statuses`. Got {}'.format(state_response_json)
+            assert len(statuses) == 1, 'Must have only one status TASK_RUNNING. Got {}'.format(state_response_json)
+            status = statuses[0]
+            container_status = status.get('container_status')
+            assert container_status
+            container_id_field = container_status.get('container_id')
+            assert container_id_field
+
+            # traverse nested container_id fields
+            container_ids = []
+            while True:
+                value = container_id_field.get('value')
+                assert value
+                container_ids.append(value)
+
+                if 'parent' not in container_id_field:
+                    break
+
+                container_id_field = container_id_field['parent']
```

Comment:
I this would be easier to read as:
```
container_ids = [container_id_field['value']]
while 'parent' in container_id_field:
    container_id_field = container_id_field['parent']
    container_ids.append[container_id_field['value']]
```